src/modules/StorageSnapper.rb

```diff
@@ -0,0 +1,151 @@
+# encoding: utf-8
+
+# Copyright (c) [2012-2015] Novell, Inc.
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may
+# find current contact information at www.novell.com.
+
+
+require "yast"
+
+
+module Yast
+
+  class StorageSnapperClass < Module
+
+
+    include Yast::Logger
+
+
+    def main
+
+      textdomain "storage"
+
+      Yast.import "String"
+      Yast.import "Installation"
+      Yast.import "InstExtensionImage"
+      Yast.import "Storage"
+
+    end
+
+
+    def BashLogOutput(command)
+      tmp = SCR.Execute(path(".target.bash_output"), command)
+
+      log.info("exit: #{tmp.fetch("exit")}")
+      tmp.fetch("stdout").each_line { |line| log.info("stdout #{line}") }
+      tmp.fetch("stderr").each_line { |line| log.info("stderr #{line}") }
+
+      return tmp.fetch("exit")
+    end
+
+
+    def ConfigureSnapper?()
+
+      part = Storage.GetEntryForMountpoint("/")
+
+      if part.fetch("used_fs", :unknown) != :btrfs
+        return false
+      end
+
+      userdata = part.fetch("userdata", {})
+      if userdata.fetch("/", "") != "snapshots"
+        return false
+      end
+
+      return true
+
+    end
+
+
+    def ConfigureSnapperStep1()
```

Comment:
well, this way there is four method step1-4 without clear requirements if it can move, if it can be called later or earlier. I think it should be at least documented, what is requirements for such calls, so if someone else want to change it in future he knows what is required and what need to be satisfied when changing.